src/main/java/org/restcomm/sdk/endpoints/RestEndpoints.java

```diff
@@ -26,6 +26,10 @@ public RestEndpoints(String url, HttpClient httpClient, Class<T> type) {
                 .defaultInstance().constructCollectionType(List.class, type));
     }
 
+    public T get(Object query) {
```

Comment:
@agafox I would probably rename this to `findSingle` to avoid conflicts with `get(id)`.
When you try to `get(Map)` with no criteria, it will force you to use an empty map. If users are careless and use `null` instead, then they may be invoking `get(id)` unwillingly. 

I would propose following naming convention: 
`List<T> getAll()`,`T get(id)`,`List<T> find(criteria)`, `T findSingle(criteria)`

Reply:
Either as you suggested like `findSingle` or findOne` whatever better works. Or `get(Optional<Object>)` works as well. I'm ok with both options.